test/e2e/generator/generate.go

```diff
@@ -73,7 +73,15 @@ func Generate(r *rand.Rand, opts Options) ([]e2e.Manifest, error) {
 	manifests := []e2e.Manifest{}
 	switch opts.P2P {
 	case NewP2PMode, LegacyP2PMode, HybridP2PMode:
+		defer func() {
+			// avoid modifying the global state.
+			original := make([]interface{}, len(testnetCombinations["p2p"]))
+			copy(original, testnetCombinations["p2p"])
+			testnetCombinations["p2p"] = original
+		}()
```

Comment:
Does this actually work as you think it should? Won't it just copy the already modified variable?

Reply:
Oh, yes, good catch, I missed that entirely. Probably only the restore wants to be deferred.